src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -80,6 +79,13 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
         {
             Stopwatch sw = Stopwatch.StartNew();
 
+            ServiceBundle.PlatformProxy.OtelInstrumentation?.LogActivity(new Dictionary<string, object> { 
+                { TelemetryConstants.MsalVersion, MsalIdHelper.GetMsalVersion() },
+                { TelemetryConstants.Platform, ServiceBundle.PlatformProxy.GetProductName() },
+                { TelemetryConstants.ClientId, AuthenticationRequestParameters.AppConfig.ClientId },
+                { TelemetryConstants.ActivityId, AuthenticationRequestParameters.RequestContext.CorrelationId }
+            });
+
             ApiEvent apiEvent = InitializeApiEvent(AuthenticationRequestParameters.Account?.HomeAccountId?.Identifier);
             AuthenticationRequestParameters.RequestContext.ApiEvent = apiEvent;
             MsalTelemetryEventDetails telemetryEventDetails = new MsalTelemetryEventDetails(TelemetryConstants.AcquireTokenEventName);
```

Comment:
Will the old ITelemetryClient logic be obsoleted or removed?

Reply:
Will be eventually in future.